types/tx.go

```diff
@@ -165,14 +166,21 @@ func ComputeProtoSizeForTxs(txs []Tx) int64 {
 // transaction wrapper, if this an be done, then it returns true. A child
 // transaction is a normal transaction that has been derived from a different
 // parent transaction. The returned hash is that of the parent transaction,
-// which allows us to remove the parent transaction from the mempool
+// which allows us to remove the parent transaction from the mempool. NOTE:
+// protobuf sometimes does not through an error if the transaction passed is not
+// a tmproto.ChildTx, since the schema for PayForMessage is kept in the app, we
+// cannot perform further checks without creating an import cycle.
 func DecodeChildTx(tx Tx) (hash []byte, unwrapped Tx, has bool) {
 	// attempt to unmarshal into a a child transaction
 	var childTx tmproto.ChildTx
 	err := proto.Unmarshal(tx, &childTx)
 	if err != nil {
 		return nil, nil, false
 	}
+	// this check isn't fool proof but it should work most of the time
+	if len(childTx.ParentTxHash) != sha256.Size {
```

Comment:
Isn't there like a constant in the code that references this already? Just in case we swap out the hash function in the future then we won't have to change many places.

Reply:
good idea, switched to using the hash size from tmhash
https://github.com/celestiaorg/celestia-core/pull/583/commits/af691ce561225d4ff1a05b5a8da0d4742e29c4aa